src/services/codefixes/fixExtendsInterfaceBecomesImplements.ts

```diff
@@ -0,0 +1,46 @@
+/* @internal */
+namespace ts.codefix {
+    registerCodeFix({
+        errorCodes: [Diagnostics.Cannot_extend_an_interface_0_Did_you_mean_implements.code],
+        getCodeActions: (context: CodeFixContext) => {
+            const sourceFile = context.sourceFile;
+            const start = context.span.start;
+            const token = getTokenAtPosition(sourceFile, start);
+
+            if (!(token.kind === SyntaxKind.Identifier && token.parent.parent.parent.kind === SyntaxKind.ClassDeclaration)) {

```

Comment:
This is admittedly pretty brittle. Changed to loop up until we find a classDecl (or not).

If we change the node upon which we report the error , then this makes it more likely that this will still work. However, this is an AST transformation (searching for existing extends clauses and removing them), so if we change the way we organize the AST, then we would need to redo this anyways.